### PR TITLE
crd: additional columns for richer information outputs in terminal.

### PR DESCRIPTION
**Pull request by SimonCqk**
e.g:

> $ kubectl get tfjobs -n kubedl
NAME    STATE     AGE   TTL-SECONDS   ACTIVE-SECONDS
mnist     Created   16d     100                       99999

### CODE REVIEW

File: config/crd/bases/xgboostjob.kubeflow.org_xgboostjobs.yaml [hunk 1]
```diff
@@ -8,6 +8,19 @@ metadata:
...
+    name: State
+    type: string
+  - JSONPath: .metadata.creationTimestamp
+    name: Age
+    type: date
+  - JSONPath: .spec.ttlSecondsAfterFinished
+    name: TTL-Seconds
+    type: integer
+  - JSONPath: .spec.activeDeadlineSeconds
+    name: Active-Seconds
```
Comment (jian-he) [inline config/crd/bases/xgboostjob.kubeflow.org_xgboostjobs.yaml]: Active-Deadline-Seconds

File: config/crd/bases/xgboostjob.kubeflow.org_xgboostjobs.yaml [hunk 2]
```diff
@@ -8,6 +8,19 @@ metadata:
...
 spec:
+  additionalPrinterColumns:
+  - JSONPath: .status.conditions[-1:].type
+    name: State
+    type: string
+  - JSONPath: .metadata.creationTimestamp
+    name: Age
+    type: date
+  - JSONPath: .spec.ttlSecondsAfterFinished
+    name: TTL-Seconds
```
Comment (jian-he) [inline config/crd/bases/xgboostjob.kubeflow.org_xgboostjobs.yaml]: use full name is better, TTL-After-Finished-Seconds

File: config/crd/bases/xgboostjob.kubeflow.org_xgboostjobs.yaml [hunk 3]
```diff
@@ -8,6 +8,19 @@ metadata:
   creationTimestamp: null
   name: xgboostjobs.xgboostjob.kubeflow.org
 spec:
+  additionalPrinterColumns:
+  - JSONPath: .status.conditions[-1:].type
+    name: State
+    type: string
+  - JSONPath: .metadata.creationTimestamp
+    name: Age
+    type: date
```
Comment (jian-he) [inline config/crd/bases/xgboostjob.kubeflow.org_xgboostjobs.yaml]: creationTimeStamp is not like age, age should be currentTime - creationTime

File: config/crd/bases/xgboostjob.kubeflow.org_xgboostjobs.yaml (hunk 1 above)
Comment (jian-he) [reply config/crd/bases/xgboostjob.kubeflow.org_xgboostjobs.yaml]: KubeBuilder has a way to generate the additionalPrint Column, we can use this
https://book.kubebuilder.io/reference/generating-crd.html#additional-printer-columns

File: config/crd/bases/xgboostjob.kubeflow.org_xgboostjobs.yaml (hunk 3 above)
Comment (SimonCqk) [reply config/crd/bases/xgboostjob.kubeflow.org_xgboostjobs.yaml:17]: the JSONPath seems do not support expressions, and `.metadata.creationTimestamp` works well, outputs of kubectl shows the duration as `currentTime - creationTime`.

File: config/crd/bases/xgboostjob.kubeflow.org_xgboostjobs.yaml (hunk 2 above)
Comment (SimonCqk) [reply config/crd/bases/xgboostjob.kubeflow.org_xgboostjobs.yaml]: done

File: config/crd/bases/xgboostjob.kubeflow.org_xgboostjobs.yaml (hunk 1 above)
Comment (SimonCqk) [reply config/crd/bases/xgboostjob.kubeflow.org_xgboostjobs.yaml]: done

**Comment by jian-he** [comment]
Could you give an example of the output ?

**Comment by SimonCqk** [comment]
$ k get tfjobs -n kubedl
NAME    STATE     AGE   TTL AFTER FINISHED SECONDS   ACTIVE DEADLINE SECONDS
mnist   Running     11h    100

@jian-he

**Pull request merged**